tests/FSharp.Compiler.ComponentTests/CompilerCompatibilityTests.fs

```diff
@@ -0,0 +1,113 @@
+module FSharp.Compiler.ComponentTests.CompilerCompatibilityTests
+
+open System
+open System.IO
+open Xunit
+open FSharp.Test
+open FSharp.Test.Assert
+open TestFramework
+
+type CompilerCompatibilityTests() =
+
+    let projectsPath = Path.GetFullPath(Path.Combine(__SOURCE_DIRECTORY__, "../projects/CompilerCompat"))
+    let libProjectPath = Path.Combine(projectsPath, "CompilerCompatLib")
+    let appProjectPath = Path.Combine(projectsPath, "CompilerCompatApp")
+    
+    let runDotnetBuild projectPath useLocalCompiler =
+        let args = 
+            if useLocalCompiler then
+                "build -c Release -p:LoadLocalFSharpBuild=True"
+            else
+                "build -c Release"
+                
+        let (exitCode, output, error) = Commands.executeProcess "dotnet" args projectPath
+        
+        if exitCode <> 0 then
+            let outputStr = String.concat "\n" output
+            let errorStr = String.concat "\n" error
+            failwith $"Build failed with exit code {exitCode}. Output: {outputStr}. Error: {errorStr}"
+        
+        String.concat "\n" output
+    
+    let runApp appBinaryPath =
+        let (exitCode, output, error) = Commands.executeProcess "dotnet" appBinaryPath (Path.GetDirectoryName(appBinaryPath))
+        (exitCode, String.concat "\n" output, String.concat "\n" error)
+    
+    let cleanBinObjDirectories projectPath =
+        let binPath = Path.Combine(projectPath, "bin")
+        let objPath = Path.Combine(projectPath, "obj")
+        
+        if Directory.Exists(binPath) then
+            Directory.Delete(binPath, true)
+        if Directory.Exists(objPath) then
+            Directory.Delete(objPath, true)
+
+    let getAppDllPath () =
+        // The app is built to artifacts directory due to Directory.Build.props
+        Path.Combine(__SOURCE_DIRECTORY__, "..", "..", "artifacts", "bin", "CompilerCompatApp", "Release", "net8.0", "CompilerCompatApp.dll")
+
+    [<Fact>]
+    member _.``Baseline scenario - Both library and app built with local compiler``() =
```

Comment:
These 3 tests could be 1 parametrized test.
You can pass version as a string.

e.g. "local" (will use local compiler via msbuild switches), "9" (will make use of global.json to enforce 9).